openpgp/packet/private_key.go

```diff
@@ -239,6 +320,11 @@ func (pk *PrivateKey) parse(r io.Reader) (err error) {
 		if len(privateKeyData) < 2 {
 			return errors.StructuralError("truncated private key data")
 		}
+
+		if pk.Version == 6 {
+			// No checksum
+			return pk.parsePrivateKey(privateKeyData)
+		} 
```

Comment:
Bit nitpicky but can we make this a negated check around the checksum code below, to deduplicate the line with the call to `parsePrivateKey`?